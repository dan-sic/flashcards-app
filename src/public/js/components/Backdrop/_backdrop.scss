.backdrop {
  display: none;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  z-index: $backdrop-z-index;

  &--show {
    display: block;
  }

  &--darken {
    background-color: rgba(0, 0, 0, 0.4);
  }
}
