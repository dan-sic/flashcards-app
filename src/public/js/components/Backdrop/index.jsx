import React from "react";

export default function Backdrop({ hideOnClick, show }) {
  const classes = show ? ["backdrop", "backdrop--show"] : ["backdrop"];

  return <div className={classes.join(" ")} onClick={hideOnClick}></div>;
}
