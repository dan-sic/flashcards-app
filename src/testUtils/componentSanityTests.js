import React from "react";
import { shallow } from "enzyme";

export const componentSanityTests = (Component, label) => {
  describe(`Sanity tests for ${label}`, () => {
    it("is defined", () => {
      expect(Component).toBeDefined();
    });

    it("renders element", () => {
      const wrapper = shallow(Component);
      expect(wrapper.isEmptyRender()).toBe(false);
    });

    it("matches the snapshot", () => {
      const wrapper = shallow(Component);
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
};
