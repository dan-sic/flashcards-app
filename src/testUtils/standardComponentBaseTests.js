import React from "react";
import { shallow } from "enzyme";

export const standardComponentBaseTests = (Component, label) => {
  describe(`Standard base test for ${label}`, () => {
    it("passes props to element", () => {
      const props = {
        onClick: jest.fn(),
        propValue: "button"
      };

      const wrapper = shallow(<Button {...props} />);
      wrapper.simulate("click");
      expect(props.onClick).toHaveBeenCalled();
      expect(
        wrapper
          .find("button")
          .first()
          .props()["type"]
      ).toEqual("button");
    });
  });
};
