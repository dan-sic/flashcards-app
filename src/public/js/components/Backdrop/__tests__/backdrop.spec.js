import React from "react";
import Backdrop from "../index";
import { shallow } from "enzyme";
import { componentSanityTests } from "../../../../../testUtils/componentSanityTests";

const foo = () => {};

describe("Backdrop", () => {
  componentSanityTests(
    <Backdrop hideOnClick={foo} show={false} darken={false} />,
    "Backdrop"
  );
  componentSanityTests(
    <Backdrop hideOnClick={foo} show={true} darken={false} />,
    "Backdrop"
  );
  componentSanityTests(
    <Backdrop hideOnClick={foo} show={false} darken={true} />,
    "Backdrop"
  );
  componentSanityTests(
    <Backdrop hideOnClick={foo} show={true} darken={true} />,
    "Backdrop"
  );

  it("hides on click", () => {
    const hideOnClick = jest.fn();
    const wrapper = shallow(
      <Backdrop hideOnClick={hideOnClick} show={true} darken={false} />
    );
    wrapper.simulate("click");
    expect(hideOnClick.mock.calls.length).toEqual(1);
  });
});
