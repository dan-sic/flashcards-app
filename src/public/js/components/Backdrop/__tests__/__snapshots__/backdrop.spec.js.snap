// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop Sanity tests for Backdrop matches the snapshot 1`] = `"<div class=\\"backdrop  \\"></div>"`;

exports[`Backdrop Sanity tests for Backdrop matches the snapshot 2`] = `"<div class=\\"backdrop backdrop--show \\"></div>"`;

exports[`Backdrop Sanity tests for Backdrop matches the snapshot 3`] = `"<div class=\\"backdrop  backdrop--darken\\"></div>"`;

exports[`Backdrop Sanity tests for Backdrop matches the snapshot 4`] = `"<div class=\\"backdrop backdrop--show backdrop--darken\\"></div>"`;
